server/app/routes/exercises.js

```diff
@@ -5,34 +5,49 @@ var Exercise = mongoose.model('Exercise');
 module.exports = router;
 
 // gets exercises and sends it to the browser
-router.get('/exercises', function(req, res) {
-    Exercise.find({}, function(err, exercises) {
-        if (!err) res.json(exercises);
-        else res.send(err);
+router.get('/', function(req, res, next) {
+    Exercise
+    .find({})
+    .exec()
+    .then(function(exercises) {
+        res.json(exercises);
+    })
+    .then(null, function(err) {
```

Comment:
This could be .then(null, next) since the second parameter is the error handler.